compaction/compaction_manager.cc

```diff
@@ -337,7 +337,7 @@ class compaction_manager::major_compaction_task : public compaction_manager::tas
     // it cannot be the other way around, or minor compaction for this table would be
     // prevented while an ongoing major compaction doesn't release the semaphore.
     virtual future<> do_run() override {
-        co_await coroutine::switch_to(_cm._compaction_sg.cpu);
+        co_await coroutine::switch_to(_cm._maintenance_sg.cpu);
 
```

Comment:
we also need to pick maintenance io class.

get_major_compaction_job() for all strategies is using service::get_local_compaction_priority()

we either add a param io group to get_major_compaction_job() like in:

compaction_strategy::get_reshaping_job(std::vector<shared_sstable> input, schema_ptr schema, const ::io_priority_class& iop, reshape_mode mode)

or just override the one returned in descriptor by get_major_compaction_job()